bootstrap.go

```diff
@@ -94,6 +106,23 @@ func GracefulDatadogShutdown() {
 	defer profiler.Stop()
 }
 
+func setConnectionType(cfg config.DatadogParameters, connectionType ConnectionType) (ConnectionType, error) {
```

Comment:
This function does not `set` anything :) 
I suggest renaming the function to `determineConnectionType` or `calculateConnectionType` instead.